.github/workflows/nightly-s3-uploads.yml

```diff
@@ -13,12 +13,13 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}
   cancel-in-progress: true
 
-jobs:
+permissions:
+  id-token: write
 
+jobs:
   upload-stats-to-s3:
     if: github.repository_owner == 'pytorch'
     runs-on: ubuntu-22.04
-    environment: upload-stats
```

Comment:
I think having an environment here might still be useful even if it doesn't have any secrets.  You could use `environment:upload-stats` in the OIDC trust role

Reply:
Changed the OIDC role in https://github.com/pytorch-labs/pytorch-gha-infra/pull/573